### PR TITLE
`start`: Harden against program interpreters that don't adhere fully to the ABI

### PR DESCRIPTION
**Pull request by alexrp**
See discussion in #20737.

### CODE REVIEW

**Comment by alexrp** [comment]
@koachan would you mind checking the sparc64 bits?

File: lib/std/start.zig [hunk 1]
```diff
@@ -291,6 +297,8 @@ fn _start() callconv(.Naked) noreturn {
             \\ mov fp, #0
             \\ mov lr, #0
             \\ mov x0, sp
+            \\ and x1, x0, #-16
+            \\ mov sp, x1
```
Comment (jacobly0) [inline lib/std/start.zig:301]:
```suggestion
            \\ and sp, x0, #-16
```
I know it's annoying to remember which parts of which aarch64 instructions can access sp :roll_eyes:

**Review by jacobly0 — CHANGES_REQUESTED** [review]
Looks good.

**Comment by alexrp** [comment]
Redid the MIPS part of the PR because some upcoming changes I'm going to make there will be bitness-specific, so merging them here is pointless. Also lowered the stack alignment value from 16 to 8 for the O32 ABI.

**Pull request merged**